code/TagField.php

```diff
@@ -310,7 +310,10 @@ protected function getTags($term) {
 		$term = Convert::raw2sql($term);
 
 		$query = $dataClass::get()
-			->filter($titleField . ':PartialMatch:nocase', $term)
+			->filter(array(
```

Comment:
Since the code recreates the query, the filter is no longer there. The code just grabs the dataClass but not any of the filter params. So yes, it picks up Tags from other blogs across the site.

So either the code needs to add the terms filter to the existing dataSource or grab the filter from the dataSource provided.

While I appreciate that it could take any kind of DataList, it seems like that's an edge case and the original desired functionality of grabbing the tags from a specific blog is not working.

Can someone guide me about how you would like it impleented:
a) Use the original datasource provided and adda filter (unsure if this will work...)
b) Check what kid of datalist is provided and if it's a Has Many then do the code provided.